docs/community-toolkit/ravendb.md

```diff
@@ -0,0 +1,299 @@
+
+Here’s an example:
+
+```csharp
+var settings = new RavenDBClientSettings
+{
+    Urls = new[] { serverUrl },
+    DatabaseName =  myDatabaseName,
+    Certificate = myCertificate
+};
+
+builder.AddRavenDBClient(settings: settings);
+```
+
+> [!Note]  
+> These methods are ideal for connecting to an existing RavenDB instance without relying on the hosting integration. This is particularly useful if you already have a standalone instance running (e.g., in the cloud) and want to connect to it using its specific details.
+
+After registration, you can retrieve the `IDocumentStore` instance and its associated `IDocumentSession` and `IAsyncDocumentSession` instances as follows:
+
+```csharp
+var documentStore = host.Services.GetRequiredService<IDocumentStore>();
+var session = host.Services.GetRequiredService<IDocumentSession>();
+var asyncSession = host.Services.GetRequiredService<IAsyncDocumentSession>();
+```
+
+### Add keyed RavenDB client
+
+If your application requires multiple `IDocumentStore` instances with different connection configurations, you can register keyed RavenDB clients using the `Microsoft.Extensions.Hosting.RavenDBClientExtension.AddKeyedRavenDBClient` extension method:
+
+```csharp
+builder.AddKeyedRavenDBClient(serviceKey: "production", connectionName: "production");
+builder.AddKeyedRavenDBClient(serviceKey: "testing", connectionName: "testing");
+```
+
+Then you can retrieve the `IDocumentStore` instances using dependency injection. For example, to retrieve the connection from an example service:
+
+```csharp
+public class ExampleService(
+    [FromKeyedServices("production")] IDocumentStore production,
+    [FromKeyedServices("testing")] IDocumentStore testing)
+{
+    // Use databases...
+}
+```
+
+For more information on keyed services, see [.NET dependency injection: Keyed services](/dotnet/core/extensions/dependency-injection#keyed-services).
+
+### Configuration 
+
+The .NET Aspire RavenDB Client integration provides multiple configuration approaches and options to meet the requirements and conventions of your project.
+
+#### Use a connection string
+
+When using a connection string from the `ConnectionStrings` configuration section, you can provide the name of the connection string when calling `builder.AddRavenDBClient`:
+
+```csharp
+builder.AddRavenDBClient("ravendb");
+```
+
+Then the connection string will be retrieved from the `ConnectionStrings` configuration section:
+
+```json
+{
+  "ConnectionStrings": {
+    "ravendb": "Url=http://localhost:8080/;Database=ravendb"
+  }
+}
+```
+
+#### Use configuration providers
+
+The .NET Aspire RavenDB integration supports <xref:Microsoft.Extensions.Configuration>. It loads the `RavenDBClientSettings` from configuration by using the `Aspire:RavenDB:Client` key. Consider the following example _appsettings.json_ that configures some of the options:
```

Comment:
... from configuration by using the ...
=>
... from the configuration using the ...